bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,114 @@
+
+
+class WeatherCog(Cog, name="Weather", description="get current weather information from a city"):
+    def __init__(self, bot):
+        self.bot = bot
+    
+    @command(name='weather', help='Show weather information in your city', usage="{city}")
+    async def weather(self, ctx, *city_input):
+        city = capwords(" ".join(city_input))
+        
+        # get current date and time from the city
+        timezone_city = getTimezone(city)
+
+        api_key = "441df3a5cadc2498e093c0367cae6817"
+        # complete_url to retreive weather info
+        complete_url = f"http://api.openweathermap.org/data/2.5/weather?appid={api_key}&q={city}"
+        response = get(complete_url)
+        data_weather = response.json()
+
+        # Now data_weather contains list of nested dictionaries, 
+        # check the value of "cod" if key is equal to 200
+        # it means the city is found otherwise, city is not found
+        if data_weather["cod"] == 200:
+            icon_id = data_weather["weather"][0]["icon"]
+            main = data_weather["main"]
+            visibility = data_weather['visibility']
+            current_temperature = main["temp"]
+            
+            kelvin_to_fahrenheit = get_kelvin_to_fahrenheit(current_temperature)
+            kelvin_to_celsius = get_kelvin_to_celsius(current_temperature)
+            
+            current_pressure = main["pressure"]
+            current_humidity = main["humidity"]
+            forcast = data_weather["weather"]
+            weather_description = forcast[0]["description"]
+            
+            embed = Embed(
+                color=self.bot.default_color,
+                title=city,
+                description=timezone_city.strftime('%m/%d/%Y %H:%M'),
+            )
+
+            embed.set_image(
+                url=f'http://openweathermap.org/img/wn/{icon_id}@2x.png'
+            )
+            embed.add_field(
+                name="Description",
+                value=capwords(weather_description),
+                inline=False
+            )
+            embed.add_field(
+                name="Visibility",
+                value=f"{visibility}m | {round(visibility*3.280839895)}ft",
+                inline=False
+            )
+            embed.add_field(
+                name="Temperature",
+                value=f"{kelvin_to_fahrenheit}°F | {kelvin_to_celsius}°C",
+                inline=False
+            )
+            embed.add_field(
+                name="Atmospheric Pressure",
+                value=f"{current_pressure} hPa",
+                inline=False
+            )
+            embed.add_field(
+                name="Humidity",
+                value=f"{current_humidity}%",
+                inline=False
+            )
+        else:
+            embed = Embed(
+                color=self.bot.default_color,
+                description=f"{city} No Found!",
+            )
+        await ctx.send(embed=embed)
+
+        
+def getTimezone(city):
```

Comment:
The three function you created, `getTimezone`, `get_kelvin_to_celcius` and `get_kelvin_to_fahrenheit` shouldn't be function. They should be methods of the WeatherCog. 

You don't need to convert them as commands, you don't need to make them async either. Just put them in the class so they become methods of the class instead.